components/brave_new_tab_ui/components/default/braveToday/customize/Discover.tsx

```diff
@@ -39,48 +39,94 @@ const Hidable = styled.div<{ hidden: boolean }>`
-    [lowerQuery, channels, showingAllCategories])
+    [channels, showingAllCategories])
 
-  return <Flex direction='column'>
-    <Header>Discover</Header>
-    <SearchInput type="search" placeholder={getLocale('braveNewsSearchPlaceholderLabel')} value={query} onChange={e => setQuery(e.currentTarget.value)} />
-    <DirectFeedResults query={query} />
-    <DiscoverSection name={getLocale('braveNewsBrowseByCategoryHeader')}>
+  return (
+    <>
+      <DiscoverSection name={getLocale('braveNewsChannelsHeader')}>
       {channels.map(c => <Hidable key={c.channelName} hidden={!visibleChannelIds.has(c.channelName)}>
         <ChannelCard key={c.channelName} channelId={c.channelName} />
       </Hidable>)}
-      {!showingAllCategories && !query && <LoadMoreButtonContainer>
+      {!showingAllCategories && <LoadMoreButtonContainer>
         <Button onClick={() => setShowingAllCategories(true)}>
-          {getLocale('braveNewsLoadMoreCategoriesButton')}
+            {getLocale('braveNewsLoadMoreCategoriesButton')}
         </Button>
       </LoadMoreButtonContainer>}
-    </DiscoverSection>
-    <DiscoverSection name={getLocale('braveNewsAllSourcesHeader')}>
-      {publishers.map(p => <Hidable key={p.publisherId} hidden={!visiblePublisherIds.has(p.publisherId)}>
-        <FeedCard publisherId={p.publisherId} />
-      </Hidable>)}
-    </DiscoverSection>
-  </Flex>
+      </DiscoverSection>
+      <DiscoverSection name={getLocale('braveNewsAllSourcesHeader')}>
+      {sortedPublishers.map(p =>
+        <Hidable key={p.publisherId} hidden={!filteredPublisherIds.includes(p.publisherId)}>
+          <FeedCard publisherId={p.publisherId} />
+        </Hidable>)}
+      </DiscoverSection>
+    </>
+  )
+}
+
+interface SearchResultsProps {
+  query: string
+}
+function SearchResults (props: SearchResultsProps) {
+  const { sortedPublishers } = useBraveNews()
+  const search = useSearch(props.query)
+  const isFetchable = (search.feedUrlQuery !== null)
+  const showFetchPermissionButton = (isFetchable && (!search.canFetchUrl || search.loading))
+
+  const hasAnyChannels = search.filteredChannels.length > 0
+  const hasAnySources = (search.filteredSources.publishers.length > 0 || search.filteredSources.direct.length > 0)
+
+  return (
+    <>
+      {hasAnyChannels &&
+      <DiscoverSection name={getLocale('braveNewsChannelsHeader')}>
+        {search.filteredChannels.map(c =>
+          <ChannelCard key={c.channelName} channelId={c.channelName} />
+        )}
+      </DiscoverSection>
+      }
+      <DiscoverSection name={getLocale('braveNewsAllSourcesHeader')}>
+        {sortedPublishers.map(r =>
+          <Hidable key={r.publisherId} hidden={!search.filteredSources.publishers.includes(r.publisherId)}>
+            <FeedCard publisherId={r.publisherId} />
+          </Hidable>
+        )}
+        {showFetchPermissionButton &&
+          <div>
+            <Button scale='tiny' onClick={() => search.setCanFetchUrl(true)} isLoading={search.loading}>
```

Comment:
This could be `search.findDirectFeeds()` if decide to use my suggestion from `useSearch`